src/run_server.sh

```diff
@@ -28,6 +28,30 @@
 # Set to `-x` for Debug logging
 set +x -o pipefail
 
+# Retrieves a setting from the server config
+function get_server_config_value() {
+    key=$1
+    sed -rn "s/^$key=(.*)$/\1/p" "$SERVER_CONFIG"
+}
+
+# Handle shutting down the server, with optional RCON quit for graceful shutdown
+function shutdown() {
+    rcon_port=$(get_server_config_value RCONPort)
+    rcon_password=$(get_server_config_value RCONPassword)
+    if [[ -n "$RCON_SHUTDOWN" ]] && \
+            [[ "$RCON_SHUTDOWN" != "false" ]] && \
+            [[ -n "$rcon_port" ]] && \
+            [[ -n "$rcon_password" ]]; then
```

Comment:
We can probably handle this check earlier in the `set_variables` function. Then we can have a more simple `rconEnabled` check here instead.